spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkStagedScan.java

```diff
@@ -77,14 +77,16 @@ public boolean equals(Object other) {
     SparkStagedScan that = (SparkStagedScan) other;
     return table().name().equals(that.table().name())
         && Objects.equals(taskSetId, that.taskSetId)
+        && readSchema().equals(that.readSchema())
         && Objects.equals(splitSize, that.splitSize)
         && Objects.equals(splitLookback, that.splitLookback)
         && Objects.equals(openFileCost, that.openFileCost);
   }
 
   @Override
   public int hashCode() {
-    return Objects.hash(table().name(), taskSetId, splitSize, splitSize, openFileCost);
+    return Objects.hash(
+        table().name(), readSchema(), taskSetId, splitSize, splitSize, openFileCost);
```

Comment:
Minor: It would be nice to keep the order of values here as the order of fields in `equals`.